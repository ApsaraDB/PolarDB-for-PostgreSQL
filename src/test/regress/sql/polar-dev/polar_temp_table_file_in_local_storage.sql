-- superuser create temp table no-use tablespace
create temp table test(id int);
select pg_relation_filepath('test') ~ 'pg_tblspc';
insert into test values(generate_series(1,10));
select * from test;
update test set id = 0 where id = 10;
truncate test;
drop table test;

-- superuser create temp table use-tablespace
show temp_tablespaces;
\! rm -rf /tmp/tablespace_temp
\! mkdir /tmp/tablespace_temp
create tablespace tbs_tmp location '/tmp/tablespace_temp';
set session temp_tablespaces = 'tbs_tmp';
show temp_tablespaces;
select pg_sleep(5);
create temp table test(id int);
select pg_relation_filepath('test') ~ 'pg_tblspc';
insert into test values(generate_series(1,10));
select * from test;
update test set id = 0 where id = 10;
truncate test;
drop table test;
drop tablespace tbs_tmp;
\! ls /tmp/tablespace_temp | wc -l
\! rm -rf /tmp/tablespace_temp

-- polar superuser create tablespace
create user temp_table_local_user with polar_superuser;
\c - temp_table_local_user
\! rm -rf /tmp/tablespace_temp
\! mkdir /tmp/tablespace_temp
create tablespace tbs_tmp location '/tmp/tablespace_temp';
drop tablespace tbs_tmp;
\! ls /tmp/tablespace_temp | wc -l
\! rm -rf /tmp/tablespace_temp
\c - postgres
drop user temp_table_local_user;