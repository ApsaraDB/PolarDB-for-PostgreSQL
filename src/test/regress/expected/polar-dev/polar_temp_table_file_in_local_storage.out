-- superuser create temp table no-use tablespace
create temp table test(id int);
select pg_relation_filepath('test') ~ 'pg_tblspc';
 ?column? 
----------
 f
(1 row)

insert into test values(generate_series(1,10));
select * from test;
 id 
----
  1
  2
  3
  4
  5
  6
  7
  8
  9
 10
(10 rows)

update test set id = 0 where id = 10;
truncate test;
drop table test;
-- superuser create temp table use-tablespace
show temp_tablespaces;
 temp_tablespaces 
------------------
 
(1 row)

\! rm -rf /tmp/tablespace_temp
\! mkdir /tmp/tablespace_temp
create tablespace tbs_tmp location '/tmp/tablespace_temp';
set session temp_tablespaces = 'tbs_tmp';
show temp_tablespaces;
 temp_tablespaces 
------------------
 tbs_tmp
(1 row)

select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

create temp table test(id int);
select pg_relation_filepath('test') ~ 'pg_tblspc';
 ?column? 
----------
 t
(1 row)

insert into test values(generate_series(1,10));
select * from test;
 id 
----
  1
  2
  3
  4
  5
  6
  7
  8
  9
 10
(10 rows)

update test set id = 0 where id = 10;
truncate test;
drop table test;
drop tablespace tbs_tmp;
\! ls /tmp/tablespace_temp | wc -l
0
\! rm -rf /tmp/tablespace_temp
-- polar superuser create tablespace
create user temp_table_local_user with polar_superuser;
\c - temp_table_local_user
\! rm -rf /tmp/tablespace_temp
\! mkdir /tmp/tablespace_temp
create tablespace tbs_tmp location '/tmp/tablespace_temp';
drop tablespace tbs_tmp;
\! ls /tmp/tablespace_temp | wc -l
0
\! rm -rf /tmp/tablespace_temp
\c - postgres
drop user temp_table_local_user;
